proof_ics23.go

```diff
@@ -166,3 +178,34 @@ func convertVarIntToBytes(orig int64, buf [binary.MaxVarintLen64]byte) []byte {
 	n := binary.PutVarint(buf[:], orig)
 	return buf[:n]
 }
+
+// GetProof gets the proof for the given key.
+func (t *ImmutableTree) GetProof(key []byte, isExist bool) (*ics23.CommitmentProof, error) {
```

Comment:
Seems weird to me that we would expose the API to take in an `isExist` bool.

Doesn't it make more sense to just check if key exists in the tree.

If it does, return an existence proof otherwise return a nonexistence proof